### PR TITLE
Update pon.lua

### PR DESCRIPTION
**Pull request by thelastpenguin**
Recent changes to garry's mod have broken pon as found by @aStonedPenguin these fixes should prevent it from breaking nutscript.

### CODE REVIEW

**Comment by thelastpenguin** [comment]
@Chessnut fairly urgent. Broke a while ago. May be causing noticeable issues.

File: gamemode/core/libs/thirdparty/sh_pon.lua [hunk 1]
```diff
@@ -45,100 +42,103 @@ _G.pon = pon;
...
-					end
-					
-				end
-			end
-		end
-		output[ #output + 1 ] = '}';
+
+
+    local first = next(tbl, nil)
+    local predictedNumeric = 1
```
Comment (thelastpenguin) [inline gamemode/core/libs/thirdparty/sh_pon.lua:69]: I kinda fucked the indentation here apparently... rip.

**Comment by thelastpenguin** [comment]
this also includes some stuff from the developmental version that does hex encoding. this is fine since you only use it for networking. The only reason it is developmental is because it may have backwards compatibility issues with data from disk though I don't think it actually does.

**Comment by thelastpenguin** [comment]
actually might have jumped the gun since stoned just found a problem with my fix.

**Review by art-gur — APPROVED** [review]

**Pull request closed without merging**